src/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -247,41 +284,122 @@ public Object getAttribute(String key, Object defaultValue) {
         return Collections.singletonList(o);
     }
 
-    public String getAttributeAsString(String key, String defaultValue) {
-        Object x = getAttribute(key);
+    /**
+     * return an attribute as a String.
+     * if given key is not found the defaultValue is returned.
+     * if the value is not a String <code> String.valueOf(x)</code> is returned.
+     * 
+     * @param key the attribute key
+     * @param defaultValue  the default value
+     * @return the attribute as a String
+     */
+    public String getAttributeAsString(final String key, final String defaultValue) {
+        final Object x = getAttribute(key);
         if ( x == null ) return defaultValue;
         if ( x instanceof String ) return (String)x;
-        return String.valueOf(x); // throws an exception if this isn't a string
+        return String.valueOf(x);
     }
 
-    public int getAttributeAsInt(String key, int defaultValue) {
-        Object x = getAttribute(key);
+    /**
+     * return an attribute as an integer.
+     * if given key is not found the defaultValue is returned.
+     * if the value is a String, the value of <code>Integer.parseInt((String)x)</code> is returned.
+     * If the value is not a String or an Integer, an exception is thrown
+     * 
+     * @param key the attribute key
+     * @param defaultValue the default value
+     *
+     * @return the attribute as an integer
+     */
+    public int getAttributeAsInt(final String key, final int defaultValue) {
+        final Object x = getAttribute(key);
         if ( x == null || x == VCFConstants.MISSING_VALUE_v4 ) return defaultValue;
         if ( x instanceof Integer ) return (Integer)x;
-        return Integer.valueOf((String)x); // throws an exception if this isn't a string
+        if ( !( x instanceof String ) ) throw new IllegalArgumentException(
+                "Cannot get attribute(key=\"" + key + ") as an integer because it's not a Number or a String. Class = " + x.getClass());
+        try {
+            return Integer.parseInt((String)x);
+        } catch (final NumberFormatException err) {
+            throw new IllegalArgumentException( "Cannot convert attribute(key=\"" + key + ") = "+ x + " as an integer.", err);
+        }
     }
-
-    public double getAttributeAsDouble(String key, double defaultValue) {
-        Object x = getAttribute(key);
+    
+    /**
+     * return an attribute as a double.
+     * if given key is not found the defaultValue is returned.
+     * if the value is as an Integer, this value is returned.
+     * if the value is a String, the value of <code>Double.parseDouble((String)x)</code> is returned.
+     * If the value is not a Double, a String or an Integer, an exception is thrown
+     * 
+     * @param key the attribute key
+     * @param defaultValue the default value
+     * @return the attribute as a double
+     */
+    public double getAttributeAsDouble(final String key, final double defaultValue) {
```

Comment:
@lbergelson what do you mean ? do you mean you don't understand the original javadoc or you don't understand my changes ?